### PR TITLE
feat(pipelines): add component_scope field to internal metrics

### PR DESCRIPTION
**Pull request by jdrouet**
Signed-off-by: Jérémie Drouet <jeremie.drouet@datadoghq.com>

### CODE REVIEW

File: src/config/id.rs [hunk 1]
```diff
@@ -13,6 +13,15 @@ pub enum ComponentScope {
     Pipeline(String),
 }
 
+impl fmt::Display for ComponentScope {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            Self::Global => write!(f, "global"),
+            Self::Pipeline(name) => write!(f, "pipeline:{}", name),
```
Comment (spencergilbert) [inline src/config/id.rs:20]: Should this `:` be a `.`? I thought we were joining the names with `.`'s now?

**Review by spencergilbert — COMMENTED** [review]
Has the new `scope` tag been added to the docs as well?

File: src/config/id.rs (hunk 1 above)
Comment (jdrouet) [reply src/config/id.rs:20]: This is to display the scope, it's not related to the `ComponentKey` itself.
With this, in the events from `internal_scope`, you'll have `{ "component_id": "foo", "component_scope": "pipeline:bar" }` so that when we introduce more scopes, we can just prefix it with the type.
Besides that, I'm ok to use `.` instead of `:`. In the previous PR, nobody raised this point (maybe nobody noted this 🤷 ) so I guess @lukesteensen and @jszwedko are ok with that 🤔

**Comment by jdrouet** [comment]
> Has the new scope tag been added to the docs as well?

Not yet, I guess it has to be done when all these will be merged.

File: src/config/id.rs (hunk 1 above)
Comment (spencergilbert) [reply src/config/id.rs:20]: Cool - I wasn't sure if the event tagging would match like `"component_scope": "pipeline.bar"`

Comment (lukesteensen) [reply src/config/id.rs:20]: The colon seems reasonable to me since it's not the name itself but the type of the scope, but I don't have a particularly strong opinion.

File: src/sources/host_metrics/mod.rs [hunk 2]
```diff
@@ -299,6 +299,7 @@ fn add_collector(collector: &str, mut metrics: Vec<Metric>) -> Vec<Metric> {
     metrics
 }
 
+#[allow(clippy::missing_const_for_fn)]
```
Comment (lukesteensen) [inline src/sources/host_metrics/mod.rs:302]: This seems unrelated. Should we make something here `const` on master? Feels a little off to add an `allow`.

Comment (jszwedko) [reply src/sources/host_metrics/mod.rs:302]: Yeah, agreed. If we need to allow it, could we add a comment about why?

Comment (jdrouet) [reply src/sources/host_metrics/mod.rs:302]: if I remove the `allow` I get
```
error: this could be a `const fn`
   --> src/sources/host_metrics/mod.rs:303:1
    |
303 | / fn init_roots() {
304 | |     #[cfg(target_os = "linux")]
305 | |     {
306 | |         use std::sync::Once;
...   |
333 | |     };
334 | | }
    | |_^
    |
note: the lint level is defined here
   --> src/lib.rs:14:9
    |
14  | #![deny(clippy::missing_const_for_fn)] // [nursery] valuable to the optimizer,
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#missing_const_for_fn

error: aborting due to previous error

error: could not compile `vector`
```

Comment (jdrouet) [reply src/sources/host_metrics/mod.rs:302]: I'll try to add `const` to the function but I think I tried and it failed to pass the tests on the CI...

Comment (jdrouet) [reply src/sources/host_metrics/mod.rs]: Ok, seems to work 🤷

Comment (jdrouet) [reply src/sources/host_metrics/mod.rs]: Fails on linux

```
   Compiling vrl-stdlib v0.1.0 (/home/ubuntu/github-actions-runner/_work/vector/vector/lib/vrl/stdlib)
error[E0013]: constant functions cannot refer to statics
Error:    --> src/sources/host_metrics/mod.rs:309:9
    |
309 |         INIT.call_once(|| {
    |         ^^^^
    |
    = help: consider extracting the value of the `static` to a `const`, and referring to that

error[E0015]: calls in constant functions are limited to constant functions, tuple structs and tuple variants
Error:    --> src/sources/host_metrics/mod.rs:309:9
    |
309 | /         INIT.call_once(|| {
310 | |             match std::env::var_os("PROCFS_ROOT") {
311 | |                 Some(procfs_root) => {
312 | |                     info!(
...   |
330 | |             }
331 | |         });
    | |__________^

error: aborting due to 2 previous errors
Error: aborting due to 2 previous errors
Some errors have detailed explanations: E0013, E0015.
For more information about an error, try `rustc --explain E0013`.
error: could not compile `vector`
Error: could not compile `vector`
To learn more, run the command again with --verbose.
make: *** [Makefile:557: test-shutdown] Error 101
Error: Process completed with exit code 2.
```

**Review by spencergilbert — APPROVED** [review]
I'll defer to the others about the const fn lint - otherwise lgtm

**Review by jszwedko — APPROVED** [review]
